src/hydrusvideodeduplicator/dedup.py

```diff
@@ -235,15 +273,15 @@ def _find_potential_duplicates(
                                     video1_hash,
                                     video2_hash,
                                     row["perceptual_hash"],
-                                    hashdb[video2_hash]["perceptual_hash"],
+                                    videos_table[video2_hash]["perceptual_hash"],
                                 )
-                                for video2_hash in islice(hashdb, row["farthest_search_index"], None)
+                                for video2_hash in islice(video_hashes, start_index, None)
                             )
 
                             # Video has now been compared against all other videos for dupes,
                             # so update farthest_search_index to the current length of the table
                             row["farthest_search_index"] = total
-                            hashdb[video1_hash] = row
+                            videos_table[video1_hash] = row
```

Comment:
@ianwal So, question here - since `videos_table` is just an in-memory copy of `hashdb`, when do the contents of `videos_table` get copied back to `hashdb`? Cause, as written, it seems to me like it never gets copied back to `hashdb`, and so you never get the benefit of actually storing the previous work done by previous runs. Or am I missing something somewhere?